core/modules/layout/layout.admin.inc

```diff
@@ -3314,6 +3314,7 @@ function layout_ajax_form_update($form, $form_state) {
 function layout_settings_page($form, &$form_state) {
   $header = array(
     'layout_template_info' => array('data' => t('Template')),
+    'description' => array('data' => t('Description')),
```

Comment:
I think `'class' => array(RESPONSIVE_PRIORITY_LOW),` would be beneficial for the description column. So this column collapses on narrow screens (like "type" already does).